### PR TITLE
Don't connect sql objects on hashCode(), equals(), or toString()

### PR DESCRIPTION
**Pull request by qualidafial**
Fixes bug identified in #707

### CODE REVIEW

File: src/test/java/org/skife/jdbi/v2/sqlobject/TestObjectMethodBehavior.java [hunk 1]
```diff
@@ -21,13 +21,15 @@
...
 /**
  * Sometimes the GC will call {@link #finalize()} on a SqlObject from
- * extremely sensitive places from within the GC machinery.  JDBI should not
- * open a {@link Connection} just to satisfy a (no-op) finalizer.
+ * extremely sensitive places from within the GC machinery.
  * <a href="https://github.com/brianm/jdbi/issues/82">Issue #82</a>.
+ * Other methods also do not use a {@link Connection} and can cause issues acquiring one.
  */
-public class TestFinalizeBehavior
+public class TestObjectMethodBehavior
```
Comment (arteam) [inline src/test/java/org/skife/jdbi/v2/sqlobject/TestObjectMethodBehavior.java]: Maybe it's a better idea to write a new test instead changing the finalize test?

File: src/main/java/org/skife/jdbi/v2/sqlobject/SqlObject.java [hunk 2]
```diff
@@ -187,9 +187,7 @@ else if (mixinHandlers.containsKey(raw_method)) {
         // this is an implicit mixin, not an explicit one, so we need to *always* add it
         handlers.putAll(CloseInternalDoNotUseThisClass.Helper.handlers());
 
-        handlers.putAll(EqualsHandler.handler());
         handlers.putAll(ToStringHandler.handler(sqlObjectType.getName()));
```
Comment (arteam) [inline src/main/java/org/skife/jdbi/v2/sqlobject/SqlObject.java]: I'm wondering why we don't remove `ToStringHandler`. Do we expect that someone could override it?

Comment (qualidafial) [inline src/main/java/org/skife/jdbi/v2/sqlobject/SqlObject.java]: Without the override, CGLIB gives it a pretty ugly toString(). There are also some tests that expect a specific toString(), so I opted not to mess with it.

File: src/test/java/org/skife/jdbi/v2/sqlobject/TestObjectMethodBehavior.java (hunk 1 above)
Comment (qualidafial) [inline src/test/java/org/skife/jdbi/v2/sqlobject/TestObjectMethodBehavior.java]: I didn't mind this rename--it just expanded the same requirement (invocation does not connect) from finalize to the other object methods. If you feel strongly about it though, I'm OK with breaking it out.

Comment (q3aiml) [inline src/test/java/org/skife/jdbi/v2/sqlobject/TestObjectMethodBehavior.java]: FWIW I took the naming more or less from jdbi3: https://github.com/jdbi/jdbi/blob/jdbi3/sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestSqlObjectMethodBehavior.java

**Comment by arteam** [comment]
Looks good to me. Let's wait for a review @stevenschlansker, then ship it.

**Comment by q3aiml** [comment]
Thank you everyone for being so responsive. I appreciate all your time and it shows in how pleasant jdbi is to use

**Pull request merged**